docs/spec/v1beta1/provider.md

```diff
@@ -730,3 +732,141 @@ kubectl create secret generic webhook-url \
 --from-literal=address="Endpoint=sb://fluxv2.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=yoursaskeygeneatedbyazure"
 ```
 
+### GitHub repository dispatch
+
+The `githubdispatch` webhook triggers a GitHub webhook event called [`repository_dispatch`](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event) for the selected repository. The `repository_dispatch` event can be used to trigger a GitHub Actions workflow with tests for the deployed service.
+
+The request includes the `event_type` and `client_payload` fields:
+
+* The `event_type` is generated by GitHub Dispatch provider by combining the Kind, Name and Namespace of the involved object in the format `{Kind}/{Name}.{Namespace}`. For example, the `event_type` for Flux configuration named `podinfo` in the `flux-system` namespace looks like this: `Kustomization/podinfo.flux-system`
+
+* The `client_payload` contains the event data from the notification controller and looks like this:
+
+```yaml
+{
+  involvedObject: {
+    apiVersion: kustomize.toolkit.fluxcd.io/v1beta2,
+    kind: Kustomization,
+    name: podinfo,
+    namespace: flux-system,
+    resourceVersion: 426573,
+    uid: b9b8554d-be26-4a3d-a97f-65f3276a097a
+  },
+  message: Deployment/podinfo/podinfo configured,
+  metadata: {
+    revision: main/96139968ca46b53462d1bf94de410a811d2026a1,
+    summary: "staging (us-west-2)"
+  },
+  reason: Progressing,
+  reportingController: kustomize-controller,
+  reportingInstance: kustomize-controller-79464d8dc5-nb9c4,
+  severity: info,
+  timestamp: 2022-04-20T12:20:28Z
+}
+```
+
+#### Setting up a GitHub workflow
+
+You need to add `repository_dispatch` trigger to trigger your GitHub workflow using the Flux notification with the relevant event type:
```

Comment:
To trigger a GitHub Actions workflow when a Flux Kustomization finishes reconciling, you need to set the event type on the `repository_dispatch` to match the Flux object ID: